google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryResultSetImpl.java

```diff
@@ -0,0 +1,554 @@
+      }
+    }
+
+    @Override
+    public int getInt(String fieldName) throws SQLException {
+      if (fieldName == null) {
+        throw new SQLException("fieldName can't be null");
+      }
+      if (cursor == null) {
+        return 0; // the column value; if the value is SQL NULL, the value returned is 0 as per
+        // java.sql.ResultSet definition
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(fieldName);
+        return fieldValue.getValue() == null ? 0 : fieldValue.getNumericValue().intValue();
+      } else { // Data received from Read API (Arrow)
+
+        Tuple<Map<String, Object>, Boolean> curTuple = (Tuple<Map<String, Object>, Boolean>) cursor;
+        if (!curTuple.x().containsKey(fieldName)) {
+          throw new SQLException(String.format("Field %s not found", fieldName));
+        }
+        Object curVal = curTuple.x().get(fieldName);
+        if (curVal == null) {
+          return 0;
+        }
+        if (curVal instanceof Text) { // parse from text to int
+          return Integer.parseInt(((Text) curVal).toString());
+        }
+        return ((BigDecimal) curVal).intValue();
+      }
+    }
+
+    @Override
+    public int getInt(int columnIndex) throws SQLException {
+      if (cursor == null) {
+        return 0; // the column value; if the value is SQL NULL, the value returned is 0 as per
+        // java.sql.ResultSet definition
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(columnIndex);
+        return fieldValue.getValue() == null ? 0 : fieldValue.getNumericValue().intValue();
+      } else { // Data received from Read API (Arrow)
+        return getInt(schemaFieldList.get(columnIndex).getName());
+      }
+    }
+
+    @Override
+    public long getLong(String fieldName) throws SQLException {
+      if (fieldName == null) {
+        throw new SQLException("fieldName can't be null");
+      }
+      if (cursor == null) {
+        return 0L; // the column value; if the value is SQL NULL, the value returned is 0 as per
+        // java.sql.ResultSet definition
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(fieldName);
+        return fieldValue.getValue() == null ? 0L : fieldValue.getNumericValue().longValue();
+      } else { // Data received from Read API (Arrow)
+        Tuple<Map<String, Object>, Boolean> curTuple = (Tuple<Map<String, Object>, Boolean>) cursor;
+        if (!curTuple.x().containsKey(fieldName)) {
+          throw new SQLException(String.format("Field %s not found", fieldName));
+        }
+        Object curVal = curTuple.x().get(fieldName);
+        if (curVal == null) {
+          return 0L;
+        } else if (curVal instanceof Long) {
+          return ((Long) curVal).longValue();
+        } else {
+          return ((BigDecimal) curVal).longValue();
+        }
```

Comment:
You may be able to check for "either Long or BigDecimal", and use their superclass of `Numeric` to call `longValue()`.

Reply:
sure, removed both and replaced with `((Number) curVal).longValue()`